### PR TITLE
Fix ancillary data links

### PR DESCRIPTION
**Pull request by alexgleith**
Using the `relative` argument makes all links relative, whether they should be or not.

This change only makes them relative if their `href` is a subfolder of the `self` `href`.

### CODE REVIEW

**Comment by alexgleith** [comment]
Sorry about formatting changes... upgraded black version.

File: apps/dc_tools/odc/apps/dc_tools/_stac.py [hunk 1]
```diff
@@ -171,19 +172,29 @@ def _get_stac_bands(
...
 
-    def _get_path(asset):
-        path = asset["href"]
+    def _get_path(asset, force_relative=False):
+        path = URL(asset["href"])
         if relative:
-            path = Path(path).name
+            try:
+                path = path.relative_to(URL(self_link).parent)
+            except (ValueError, TypeError):
```
Comment (Kirill888) [inline apps/dc_tools/odc/apps/dc_tools/_stac.py]: `TypeError` looks suss? when does this happen?

Comment (Kirill888) [reply apps/dc_tools/odc/apps/dc_tools/_stac.py]: if it's when `self_link` is `None` then you should check that before attempting to compute parent of that link.

File: apps/dc_tools/odc/apps/dc_tools/_stac.py [hunk 2]
```diff
@@ -171,19 +172,29 @@ def _get_stac_bands(
...
+                if force_relative:
+                    path = path.name
```
Comment (Kirill888) [inline apps/dc_tools/odc/apps/dc_tools/_stac.py:194]: Should have at least some comment describing what happens and in what situations. I can only guess from looking? If asset doesn't share same path with the stac item, just use filename as relative path?

File: apps/dc_tools/odc/apps/dc_tools/_stac.py (hunk 1 above)
Comment (alexgleith) [reply apps/dc_tools/odc/apps/dc_tools/_stac.py]: Cleaned this up to make it more obviousl

File: apps/dc_tools/odc/apps/dc_tools/_stac.py (hunk 2 above)
Comment (alexgleith) [reply apps/dc_tools/odc/apps/dc_tools/_stac.py:194]: On it.

**Review by Kirill888 — APPROVED** [review]

**Pull request merged**